controllers/semeru_compiler.go

```diff
@@ -367,3 +371,28 @@ func getSemeruJavaOptions(instance *wlv1.WebSphereLibertyApplication) []string {
 	}
 	return nil
 }
+func (r *ReconcileWebSphereLiberty) areSemeruCompilerResourcesReady(wlva *wlv1.WebSphereLibertyApplication) error {
+	var replicas, readyReplicas, updatedReplicas int32
+	namespacedName := types.NamespacedName{Name: wlva.GetName() + SemeruLabelNameSuffix, Namespace: wlva.GetNamespace()}
+
+	// Check if deployment exists
+	deployment := &appsv1.Deployment{}
+	err := r.GetClient().Get(context.TODO(), namespacedName, deployment)
+	if err != nil {
+		return errors.New("semeru cloud compiler is not ready: deployment is not created")
+	}
+
+	// Get replicas
+	er := int32(1)
+	expectedReplicas := &er
+	ds := deployment.Status
+	replicas, readyReplicas, updatedReplicas = ds.Replicas, ds.ReadyReplicas, ds.UpdatedReplicas
+
+	// Check if all replicas are equal to the expected replicas
+	if replicas == *expectedReplicas && readyReplicas == *expectedReplicas && updatedReplicas == *expectedReplicas {
+		return nil // Semeru ready
+	} else if replicas > *expectedReplicas {
+		return errors.New("semeru cloud compiler is not ready: replica set is progressing")
```

Comment:
Confirmed that WebSphere Automation uses capitalization and punctuation without issues, so we can do the same, so the messages look proper.